### DISCUSSION TITLE
I had a question about the controller support for devilutionX

### DISCUSSION THREAD

**Discussion by dmanfx123** [diasurgical/DevilutionX#2093] [Ideas]
I'm not a coder, I have plenty experience installing and using mods, so this is still coming from a laymen. I recently discovered this version of Diablo as I was looking for a version that  both supports 16:9 and higher fps (achieved using Fluffymod), but still plays more like vanilla Diablo 1. Nothing crazy like belzebub or The Hell 1 or 2. I discovered watching a video of a DevilutionX handheld port that it does have controller support. So I plugged in one of mine to try it out. The one thing I noticed was that there was no standing attack button. It relies entirely on an auto targeting system. So you guys managed to make picking up items, a left mouse button task, and attacking, also left mouse button, into separate buttons, and I'm guessing this was necessary because of how the targeting system needed to be set up. What I'm wondering is, why there isn't a shift "stand your ground" button, or even a button that combines shift and left mouse button. Is this just impossible on the current game? I know attacking while holding shift and aiming in the direction an enemy will still hit them. It'd be nice if there was a button that used the standing attack for manual arrow shots and melee. Apologies if this has been mentioned elsewhere. Thanks again for making this thing, Diablo 1 deserves a remaster as much as 2.

**Comment by julealgon** [comment]
> Is this just impossible on the current game?

There is no such limitation. The main issue with the controller IIRC from discussions is that there are few buttons left to use (if any) and the game targets a fairly limited set of buttons so it works with hardware with fewer buttons.

This has been discussed quite a bit in the discord server, so I suggest you join there if you wanna take a look. It is definitely possible to implement shift+attack and there are some open issues/PRs related to this already.

**Reply by dmanfx123** [reply]
Alright, I'll check it out. I wonder if it could be set up so that both melee and ranged default to shift attack, but then have it so bows use lock on targeting and melee doesn't, that way the bows can still lock on when an enemy is onscreen and then you could still fire at will when they're off. *edit* Any idea which server I'd find this discussion in? I don't have permission to post in any of the devilution servers. Not super familiar with how discord works. Thanks.

**Reply by qndel** [reply]
There's only one devilution server, you can find the link in readme

**Reply by dmanfx123** [reply]
Hmm. I found a link to it on the main page here on github for devlutionx, but the one in the read me says the invite is invalid.

**Reply by qndel** [reply]
https://discord.gg/YQKCAYQ

**Reply by AJenbo** [reply]
https://github.com/diasurgical/devilutionX#readme

**Reply by julealgon** [reply]
> but the one in the read me says the invite is invalid.

The one in the readme is invalid. I think it has been corrected since.

**Reply by dmanfx123** [reply]
Ahhh okay. I got it now. I was misreading. It said I didn't have permission to see message history, I thought it was just overall permission to post. Thanks guys.
